ipv8/attestation/trustchain/database.py

```diff
@@ -32,8 +33,18 @@ def __init__(self, working_directory, db_name):
         super(TrustChainDB, self).__init__(db_path)
         self._logger.debug("TrustChain database path: %s", db_path)
         self.db_name = db_name
+        self.listeners_map = block_listeners
```

Comment:
This class should not care about listeners, just the `block_type`->`block_class` mapping. You can add these to the persistence as they are registered in the [TrustChain community](https://github.com/Tribler/py-ipv8/blob/master/ipv8/attestation/trustchain/community.py#L89).